src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -15,13 +29,20 @@ namespace NUnit.Framework.Internal.Execution
     /// or to queue them for forwarding on another thread or at
     /// a later time.
     /// </summary>
-    public abstract class Event
+    public abstract class Event : IEvent<ITestListener>, IDisposable
     {
         /// <summary>
         /// The Send method is implemented by derived classes to send the event to the specified listener.
         /// </summary>
         /// <param name="listener">The listener.</param>
         public abstract void Send(ITestListener listener);
+
+        /// <summary>
+        /// Not quite sure why this is needed, but the compiler complained.
```

Comment:
It is needed because the class is declared `IDisposable`. Like the `Send` method as a minimum you need an `abstract` one.

Reply:
Yes, I that was added AFTER I got a compilation error.  
Without it, I get this:
![image](https://github.com/nunit/nunit/assets/203432/c58dd348-86e9-4b7d-b260-bf35829c0a9c)

Suggestions?

Reply:
In the EventPump declaration you put the `IDisposable` on the `where` clause instead of the class itself. You need to specify interface before where clause

Reply:
Ahhrrg!!   Thanks @manfred-brands !   That was indeed the case!  I was surely blinded there! :-)